engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/widget/ContentListHelper.java

```diff
@@ -252,23 +254,18 @@ protected boolean extractOrCategoryFilterClause(ApsProperties config) {
         return categoryCodes;
     }
 
-    @Deprecated
-    protected void addShowletFilters(IContentListTagBean bean, ApsProperties showletParams, String showletParamName, RequestContext reqCtx) {
-        this.addWidgetFilters(bean, showletParams, showletParamName, reqCtx);
-    }
-
-    protected void addWidgetFilters(IContentListTagBean bean, ApsProperties widgetParams, String widgetParamName, RequestContext reqCtx) {
+    protected EntitySearchFilter[] createWidgetFilters(IContentListTagBean bean, ApsProperties widgetParams, String widgetParamName, RequestContext reqCtx) {
         if (null == widgetParams) {
-            return;
+            return bean.getFilters();
```

Comment:
This looks like a pretty big change possibly. Given that nothing would get set in the old call. Is this covered by a test and do we know that the contract is correct if `widgetParams` is null?

Reply:
The signature of the method has changed; before the modification the method added the filters into the "bean" object (erroneously changing the key with which the list was searched in cache), now it creates and returns a list of filters without altering the "bean" object.
BTW, I added other junit tests in order to cover other cases.